src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java

```diff
@@ -88,8 +92,36 @@ private void applyProjectPlugin(Project project) {
         });
     }
 
+    private void applyProjectPluginGradle4(Project project) {
+        if (!project.equals(project.getRootProject())) {
+            throw new GradleException("Common custom user data plugin may only be applied to root project");
+        }
+        project.getPluginManager().withPlugin("com.gradle.build-scan", __ -> {
+            CustomGradleEnterpriseConfig customGradleEnterpriseConfig = new CustomGradleEnterpriseConfig();
+
+            BuildScanExtension buildScan = project.getExtensions().getByType(BuildScanExtension.class);
+            customGradleEnterpriseConfig.configureGradleEnterprise(buildScan);
```

Comment:
Getting by class name will not work in all cases, we need to get by name and use the proxy.

Reply:
Changed.